src/materials/tdycharacteristiccurves.c

```diff
@@ -504,20 +505,46 @@ void RelativePermeability_Mualem_Unsmoothed(PetscReal m,PetscReal Se,PetscReal *
 static PetscErrorCode CubicPolynomialSetup(PetscReal x1, PetscReal x2,
                                            PetscReal rhs[4]) {
 
-  PetscInt n = 4, nrhs = 1;
+  PetscInt N = 4, d;
+  PetscInt *indx;
+  PetscReal **AA;
+  PetscReal *rhs_values;
+  PetscErrorCode ierr;
+
+  ierr = TDyAllocate_RealArray_2D(&AA,N,N); CHKERRQ(ierr);
```

Comment:
These need to be deallocated at the end of the function to avoid a memory leak. There are other ways to allocate "work arrays" in C (even multidimensional ones). Maybe we should chat about that when it's convenient.